nilearn/interfaces/bids/_utils.py

```diff
@@ -0,0 +1,127 @@
+"""Utility functions for the nilearn.interfaces.bids module."""
+import json
+import warnings
+
+import nilearn
+
+
+def _clean_contrast_name(contrast_name):
+    """Remove prohibited characters from name and convert to camelCase.
+
+    BIDS filenames, in which the contrast name will appear as a
+    contrast-<name> key/value pair, must be alphanumeric strings.
+
+    Parameters
+    ----------
+    contrast_name : :obj:`str`
+        Contrast name to clean.
+
+    Returns
+    -------
+    new_name : :obj:`str`
+        Contrast name converted to alphanumeric-only camelCase.
+    """
+    new_name = contrast_name[:]
+
+    # Some characters translate to words
+    new_name = new_name.replace('-', ' Minus ')
+    new_name = new_name.replace('+', ' Plus ')
+    new_name = new_name.replace('>', ' Gt ')
+    new_name = new_name.replace('<', ' Lt ')
+
+    # Others translate to spaces
+    new_name = new_name.replace('_', ' ')
+
+    # Convert to camelCase
+    new_name = new_name.split(' ')
+    new_name[0] = new_name[0].lower()
+    new_name[1:] = [c.title() for c in new_name[1:]]
+    new_name = ' '.join(new_name)
+
+    # Remove non-alphanumeric characters
+    new_name = ''.join(ch for ch in new_name if ch.isalnum())
+
+    # Let users know if the name was changed
+    if new_name != contrast_name:
+        warnings.warn(
+            f'Contrast name "{contrast_name}" changed to "{new_name}"'
+        )
+    return new_name
+
+
+def _generate_model_metadata(out_file, model):
+    """Generate a sidecar JSON file containing model metadata."""
+    # Define which FirstLevelModel attributes are BIDS compliant and which
+    # should be bundled in a new "ModelParameters" field.
+    DATA_ATTRIBUTES = [
+        't_r',
+    ]
+    PARAMETER_ATTRIBUTES = [
+        'drift_model',
+        'hrf_model',
+        'standardize',
+        'high_pass',
+        'target_shape',
+        'signal_scaling',
+        'drift_order',
+        'scaling_axis',
+        'smoothing_fwhm',
+        'target_affine',
+        'slice_time_ref',
+        'fir_delays',
+    ]
+    ATTRIBUTE_RENAMING = {
+        't_r': 'RepetitionTime',
```

Comment:
Is this a BIDS requirement ? If yes, we should rename the parameter across the codebase.

Reply:
I'm not sure if RepetitionTime is a required metadata field for GLM outputs, but, if it is included, it should be called RepetitionTime. I erred on the side of including everything I could infer from the available parameters and data, but we may want to be more selective about what metadata we carry on to the sidecar files, if only because some of it could be extraneous.